sympy/solvers/solveset.py

```diff
@@ -1133,13 +1134,155 @@ def _is_exponential(f, symbol):
     The function extracts each term of the equation and checks if it is
     of exponential form w.r.t `symbol`.
     """
+    rv = False
+    for expr_arg in _term_factors(f):
+        if symbol not in expr_arg.free_symbols:
+            continue
+        if (isinstance(expr_arg, Pow) and
+           symbol not in expr_arg.base.free_symbols or
+           isinstance(expr_arg, exp)):
+            rv = True  # symbol in exponent
+        else:
+            return False  # depended on symbol in non-exponential way
+    return rv
+
+
+def _solve_logarithm(lhs, rhs, symbol):
+    r"""
+    Helper to solve logarithmic equations which are reducible
+    to a single instance of `log`.
+
+    Logarithmic equations are (currently) the equations that contains
+    `log` terms which can be reduced to a single log term or
+    a constant using various logarithmic identities.
+
+    For example:
+
+    .. math:: log(x) + log(x - 4)
+
+    can be reduced to:
+
+    .. math:: log(x(x - 4))
+
+    Parameters
+    ==========
+
+    f : Expr
```

Comment:
`f` is not a parameter to the function, instead `lhs` and `rhs` are.

Reply:
This has been addressed.